### ISSUE TITLE
Issue in scrolling if the index remains the same

### ISSUE THREAD

**Issue by jagan999** [marcglasberg/indexed_list_view#2]
Hello,
This code works awesomely well except I hit another issue today. I am using scrollToIndex when I go from a list screen to edit a record's details and then return back. Now if I keep editing the same record, the second time I return to the list screen, it scrolls to the first row (your code senses that there's no change in the index and if so, scrolls to 0.0) this is annoying. What happens if this check is removed in your code?
Thanks

**Comment by jagan999** [comment]
I tried commenting out this if condition - if (_startIndex != index)
The code seemed to work fine with no issues. Pl check if this can be commented out with no side effects.

**Comment by marcglasberg** [comment]
Commenting it out will have the side effect of unnecessarily rebuilding widgets under certain circumstances. Please help me understand the problem. Could you modify the example code (and post it here) so that I can see the problem myself?

**Comment by jagan999** [comment]
Hello
Unfortunately my app code is too large to be shared. And it's no longer a test code. I'll see if I can create some snippets so that you know what I'm saying.
Thanks

**Comment by jagan999** [comment]
Hello,
I pulled in your latest code and see that you've made some changes that I had tried out. The minScrollExtent being set to 0.0 does not seem to have any negative effect. The other change in animateTo where it doesn't do a check for _startIndex != index also seems to work fine. Thanks for those.

However, I'm finding a baffling problem that I probably didn't notice earlier. Here's a snippet of code from my app that is used for listing entities in my app. I'm noticing the below behavior and I'm unable to explain why the code works this way.

a. Case where it works fine: If I select a particular row and go to the edit screen, then return to the listing, the correct matchedIndex is printed and the list scrolls to the matchedIndex.
b. Case where it does not work fine: Instead of editing an existing row, if I click a FAB to create a new entity, then return to the list, the correct matchedIndex of the newly created row is printed BUT the list stays where it was previously. It DOES NOT scroll to the matchedIndex. Now, if I edit this newly created row (basically the same matchedIndex) and then return to the list, it scrolls to this row properly. 

Any clues?

---snippet below---

class EntitiesPageState extends State<EntitiesPage> {
  SearchBar searchBar;
  String lastSearch = "";
  List<Entity> _items = List(), _nullsearchitems = List();
  var selectedEntity;
  IndexedScrollController _scrollController = IndexedScrollController();
  final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
  bool displaySortFilter = true;
  bool isLoading = true;
  String customFilter, activestatusFilter, currentFilter, currentSort;

  @override
  void initState() {
    super.initState();
    refreshList();
  }

  Future<Null> refreshList() async {
    isLoading = true;
    await widget.entityModel
        .list('', currentFilter, activestatusFilter, currentSort)
        .then((entities) {
      setState(() {
        _nullsearchitems = entities;
        _items = entities;
        isLoading = false;
      });
    });

    if (_items.length > 0) {
      int matchedIndex = _items.indexOf(selectedEntity);
      print('selectedEntity = ${selectedEntity.toString()} \nmatchedIndex = $matchedIndex');
      if(matchedIndex != -1)
        _scrollController.animateToIndex(matchedIndex, duration: Duration(seconds: 2), curve: Curves.ease);
    }
  }
